### PR TITLE
JWTRetrievalStrategy

### PR DESCRIPTION
**Pull request by AlexDpy**

### CODE REVIEW

File: src/Providers/Silex/SecurityJWTServiceProvider.php [hunk 1]
```diff
@@ -20,18 +21,35 @@ class SecurityJWTServiceProvider implements ServiceProviderInterface
...
     public function register(Application $app)
     {
+        $app['security.jwt_retrieval.authorization_bearer.strategy'] = $app->share(function () {
+            return new AuthorizationBearerStrategy();
+        });
+
+        $app['security.jwt_retrieval.query_parameter.strategy'] = $app->share(function () {
+            return new QueryParameterStrategy();
+        });
+
```
Comment (remi-san) [inline src/Providers/Silex/SecurityJWTServiceProvider.php:32]: add the chained one also, as it can be usefull...

Comment (AlexDpy) [inline src/Providers/Silex/SecurityJWTServiceProvider.php:32]: done

File: src/Providers/Silex/SecurityJWTServiceProvider.php [hunk 2]
```diff
@@ -20,18 +22,42 @@ class SecurityJWTServiceProvider implements ServiceProviderInterface
...
                 $converter = new SecurityUserConverter();
                 $userBuilder = new JWTUserBuilder($decoder, $encoder, $converter);
+
                 return new JWTAuthenticationProvider($userBuilder);
             });
 
-            $app['security.authentication_listener.' . $name . '.jwt'] = $app->share(function () use ($app) {
-                return new JWTListener($app['security.token_storage'], $app['security.authentication_manager']);
+            $app['security.authentication_listener.' . $name . '.jwt'] = $app->share(function () use ($app, $name) {
+                $strategyName = isset($app['security.jwt_retrieval.strategy_name'])
```
Comment (remi-san) [inline src/Providers/Silex/SecurityJWTServiceProvider.php]: shouldn't it be passed through $options instead of $app?

Comment (AlexDpy) [inline src/Providers/Silex/SecurityJWTServiceProvider.php]: You're right. Done :)

**Pull request merged**